users/stdiscosrv.rst

```diff
@@ -346,6 +346,121 @@ page. Note that that page is directed at setting up a proxy for the
 Syncthing web UI. You should do the proper path and port adjustments to proxying
 the discovery server and your particular setup.
 
+Traefik 2.5+
+""""""""""""
+
+Traefik will send ``X-Forwarded-For`` by default.
+
+To send ``X-Forwarded-Tls-Client-Cert``:
+
+#. Add a `TLS Options <https://doc.traefik.io/traefik/https/tls/#client-authentication-mtls>`_
+   that sets ``clientAuthType`` to ``RequireAnyClientCert`` in your dynamic
+   config. Without this, Traefik will not pass the client certificate in the
+   ``X-Forwarded-Tls-Client-Cert`` header.
+
+   .. code-block:: yaml
+
+     tls:
+       options:
+         syncthing-discosrv:
+           clientAuth:
+             clientAuthType: RequireAnyClientCert
+
+#. Also in your dynamic config, add a middleware to pass the TLS client
+   cert (`passtlsclientcert.pem=true`), add that to the router, and set your
+   custom TLS options.
+
+   .. code-block:: yaml
+
+     syncthing-discosrv:
+       image: syncthing/discosrv
+       volumes:
+         - ./syncthing/DATA/syncthing-discosrv:/var/stdiscosrv
+       labels:
+         - "traefik.enable=true"
+         - "traefik.http.middlewares.syncthing-discosrv-middleware.passtlsclientcert.pem=true"
+         - "traefik.http.services.syncthing-discosrv.loadbalancer.server.port=8443"
+         - "traefik.http.routers.syncthing-discosrv.entrypoints=https"
+         - "traefik.http.routers.syncthing-discosrv.rule=Host(`st-ds.xxx.dev`)"
+         - "traefik.http.routers.syncthing-discosrv.tls.options=syncthing-discosrv@file"
+         - "traefik.http.routers.syncthing-discosrv.middlewares=syncthing-discosrv-middleware"
+       command:
+         - "-http"
+
+To send ``X-Client-Port``:
+
+Traefik has ``X-Forwarded-Port`` which can only be changed using a
+plugin (plugins were added in version 2.5 of Traefik).
```

Comment:
For what it's worth I think this sections is wrong. `X-Forwarded-Port` is the port number on the proxy that received the request, i.e., most likely `443`. The `X-Client-Port` we want is the source port on the client side. If that's not available that's mostly fine, and this section can be skipped.